x/tokenfactory/keeper/create_denom.go

```diff
@@ -0,0 +1,101 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+
+	"github.com/titantkx/titan/x/tokenfactory/types"
+)
+
+// ConvertToBaseToken converts a fee amount in a whitelisted fee token to the base fee token amount
+func (k Keeper) CreateDenom(ctx sdk.Context, creatorAddr string, subdenom string) (newTokenDenom string, err error) {
+	denom, err := k.validateCreateDenom(ctx, creatorAddr, subdenom)
+	if err != nil {
+		return "", err
+	}
+
+	err = k.chargeForCreateDenom(ctx, creatorAddr)
+	if err != nil {
+		return "", err
+	}
+
+	err = k.createDenomAfterValidation(ctx, creatorAddr, denom)
+	return denom, err
+}
+
+// Runs CreateDenom logic after the charge and all denom validation has been handled.
+// Made into a second function for genesis initialization.
+func (k Keeper) createDenomAfterValidation(ctx sdk.Context, creatorAddr string, denom string) (err error) {
+	_, exists := k.bankKeeper.GetDenomMetaData(ctx, denom)
+	if !exists {
+		denomMetaData := banktypes.Metadata{
+			DenomUnits: []*banktypes.DenomUnit{{
+				Denom:    denom,
+				Exponent: 0,
+			}},
+			Base:    denom,
+			Name:    denom,
+			Symbol:  denom,
+			Display: denom,
+		}
+
+		k.bankKeeper.SetDenomMetaData(ctx, denomMetaData)
+	}
+
+	authorityMetadata := types.DenomAuthorityMetadata{
+		Admin: creatorAddr,
```

Comment:
Does it safe when allow Admin of token free to set metadata of token?

What if someone create a token and then try to fake USDC token

Reply:
Token metadata is only for displaying purpose. User should be able to verify token by denom name